security/pkg/pki/util/keycertbundle_test.go

```diff
@@ -343,145 +343,131 @@ func TestNewVerifiedKeyCertBundleFromFile(t *testing.T) {
 
 // Test the root cert expiry timestamp can be extracted correctly.
 func TestExtractRootCertExpiryTimestamp(t *testing.T) {
-	t0 := time.Now()
-	cert, key, err := GenCertKeyFromOptions(CertOptions{
-		Host:         "citadel.testing.istio.io",
-		NotBefore:    t0,
-		TTL:          time.Minute,
-		Org:          "MyOrg",
-		IsCA:         true,
-		IsSelfSigned: true,
-		IsServer:     true,
-		RSAKeySize:   2048,
-	})
-	if err != nil {
-		t.Errorf("failed to gen cert for Citadel self signed cert %v", err)
-	}
-	kb, err := NewVerifiedKeyCertBundleFromPem(cert, key, nil, cert)
-	if err != nil {
-		t.Errorf("failed to create key cert bundle: %v", err)
-	}
 	testCases := []struct {
-		name string
-		ttl  float64
-		time time.Time
+		notBefore time.Time
```

Comment:
We should have names for our tests imo; especially when magic numbers are in play. It's not immediately obvious to me the significance of the numbers and what exactly they represent in the context of this test.

Same for other test cases defined in this PR.